### ISSUE TITLE
Master/Detail does not render custom components for 'detail' table.

### ISSUE THREAD

**Issue by LinnykS** [ag-grid/ag-grid#2371]
**I'm submitting a ...**  (check one with "x")
```
[x] bug report => search github for a similar issue or PR before submitting
[ ] feature request
[ ] support request => Please do not submit support request here, instead see https://github.com/ceolter/ag-grid/blob/master/CONTRIBUTING.md#question
```

**Current behavior**

Master/Detail feature seems has an issue with rendering custom components for 'detail' table. 
It perfectly fine renders components for the first column (despite on 'cellRenderer' or 'template' field is used) but for the next columns accept just simple text or direct HTML templates. Note: works fine for 'master' table.

**Expected behavior**

Master/Detail -> 'detail' grid has the ability renders custom components for cells, according to documentation note: 

> The detail grid is a fully featured instance of ag-Grid, so any configuration can be set on the detail grid that you would set any other grid.

**Minimal reproduction of the problem with instructions**

Please, take a look at the attachment. No any error messages in console.
![master_detail_custom_rendering_issue](https://user-images.githubusercontent.com/4090548/39812308-d0b2e34e-5394-11e8-8b0e-ba0a90dd665f.png)

Demo:
![demo](https://user-images.githubusercontent.com/4090548/39813869-dca0fe06-539a-11e8-8c94-e218e0291ac4.png)

**What is the motivation / use case for changing the behavior?**

This feature pretty good could cover my project needs but unfortunately, I cannot use it because of this issue.

**Please tell us about your environment:**

WebStrom 11.0.4

* **ag-Grid version:** 17.1.1

* **Browser:** Chrome

* **Language:** ES6 / AngularJS (1.5.8)

**Comment by bimalpaul** [comment]
Hi @LinnykS , would you be willing to share your controller for the angular JS grid with master-detail? I cannot for whatever reason get the master detail to work at all. All other features seem to work just fine!

**Comment by LinnykS** [comment]
Hey @bimalpaul. Everything from documentation actually. I've just created the simple instance of a grid and the appropriate options for it:
`this.gridOptions = {
            columnDefs: mainColumnDefs,
            masterDetail: true,
            context: this,
            angularCompileRows: true,
            detailCellRendererParams: {
                detailGridOptions: {
                    columnDefs,
                    getDetailRowData: (params) => { params.successCallback(params.data.projects); }
            }
        };`
Where:
-`mainColumnDefs` = coulmn settings for Master table
-`columnDefs` = coulmn settings for Detail table

**Comment by nelkenson** [comment]
I logged a similar/same issue at the ag-grid-enterprise project: https://github.com/ag-grid/ag-grid-enterprise/issues/182. Here's a plunker demonstrating the problem https://plnkr.co/edit/LFCwAnSYDo0zmBhbCveb

**Comment by LinnykS** [comment]
@nelkenson If you delete this option from details grid options it will be working fine and even render first column (as I mentioned in description) but that's all.

**Comment by nelkenson** [comment]
I agree. Unfortunately, we need angularCompileRows turned on for the details grid.

**Comment by AlvaroDDG86** [comment]
I have the same problem, does anyone know the solution?

**Comment by nelkenson** [comment]
Here's a plunker demonstrating the problem https://plnkr.co/edit/LFCwAnSYDo0zmBhbCveb

It appears to be directly related to having angularCompileRows set to true in the details grid.
